cedar-policy-validator/src/schema.rs

```diff
@@ -2328,6 +2328,219 @@ mod test {
                     .build());
         });
     }
+
+    #[test]
+    fn test_common_type_name_conflicts() {
+        let src: serde_json::Value = json!({
+            "": {
+                "commonTypes": {
+                    "Record": {
+                        "type": "Record",
+                        "attributes": {
+                            "a": {
+                                "type": "Long",
+                            }
+                        }
+                    }
+                },
+                "entityTypes": {
+                    "b": {
+                        "shape" : {
+                            "type" : "Record",
+                            "attributes" : {
+                                "c" : {
+                                    "type" : "Record"
```

Comment:
IDK if I'm a fan of allowing this.

Reply:
I agree it looks indeed weird but we allow it in the human-readable format. I think we should minimize the differences. In the meanwhile, it doesn't hurt to have it?

Reply:
yeah, I'm not opposed to it since it's what the Cedar schema format does, but it was definitely intentional that this was an error in the initial JSON schema parser.